bundle/phases/deploy.go

```diff
@@ -19,55 +19,90 @@ import (
 	"github.com/databricks/cli/bundle/scripts"
 	"github.com/databricks/cli/libs/cmdio"
 	terraformlib "github.com/databricks/cli/libs/terraform"
+	tfjson "github.com/hashicorp/terraform-json"
 )
 
-func approvalForUcSchemaDelete(ctx context.Context, b *bundle.Bundle) (bool, error) {
-	tf := b.Terraform
-	if tf == nil {
-		return false, fmt.Errorf("terraform not initialized")
-	}
-
-	// read plan file
-	plan, err := tf.ShowPlanFile(ctx, b.Plan.Path)
-	if err != nil {
-		return false, err
-	}
-
-	actions := make([]terraformlib.Action, 0)
-	for _, rc := range plan.ResourceChanges {
-		// We only care about destructive actions on UC schema resources.
-		if rc.Type != "databricks_schema" {
+func parseTerraformActions(changes []*tfjson.ResourceChange, toInclude func(typ string, actions tfjson.Actions) bool) []terraformlib.Action {
+	res := make([]terraformlib.Action, 0)
+	for _, rc := range changes {
+		if !toInclude(rc.Type, rc.Change.Actions) {
 			continue
 		}
 
 		var actionType terraformlib.ActionType
-
 		switch {
 		case rc.Change.Actions.Delete():
 			actionType = terraformlib.ActionTypeDelete
 		case rc.Change.Actions.Replace():
 			actionType = terraformlib.ActionTypeRecreate
 		default:
-			// We don't need a prompt for non-destructive actions like creating
-			// or updating a schema.
+			// No use case for other action types yet.
 			continue
 		}
 
-		actions = append(actions, terraformlib.Action{
+		res = append(res, terraformlib.Action{
 			Action:       actionType,
 			ResourceType: rc.Type,
 			ResourceName: rc.Name,
 		})
 	}
 
-	// No restricted actions planned. No need for approval.
-	if len(actions) == 0 {
+	return res
+}
+
+func approvalForDeploy(ctx context.Context, b *bundle.Bundle) (bool, error) {
+	tf := b.Terraform
+	if tf == nil {
+		return false, fmt.Errorf("terraform not initialized")
+	}
+
+	// read plan file
+	plan, err := tf.ShowPlanFile(ctx, b.Plan.Path)
+	if err != nil {
+		return false, err
+	}
+
+	schemaActions := parseTerraformActions(plan.ResourceChanges, func(typ string, actions tfjson.Actions) bool {
+		// Filter in only UC schema resources.
+		if typ != "databricks_schema" {
+			return false
+		}
+
+		// We only display prompts for destructive actions like deleting or
+		// recreating a schema.
+		return actions.Delete() || actions.Replace()
+	})
+
+	dltActions := parseTerraformActions(plan.ResourceChanges, func(typ string, actions tfjson.Actions) bool {
+		// Filter in only DLT pipeline resources.
+		if typ != "databricks_pipeline" {
+			return false
+		}
+
+		// Recreating DLT pipeline leads to metadata loss and for a transient period
+		// the underling tables will be unavailable.
+		return actions.Replace()
```

Comment:
What about `actions.Delete()`?

Reply:
It's out of scope for this PR for now. It does make sense to me and I think we should do it. We regularly hear reports (one just today) of customers accidentally deleting their DLT pipelines.

I'd like to see what @pietern 's thoughts are regarding this.

Reply:
As discussed offline, yes, we should still have this prompt for deletion.